build-image/Dockerfile

```diff
@@ -0,0 +1,89 @@
+FROM docker.io/bitnami/minideb:bullseye-amd64 as slimmed-rust
+
+# The rust version to use
+# The 1.69 toolchain is temporarily required to build ink! contracts because of
+# https://github.com/paritytech/cargo-contract/issues/1139
+ARG RUST_VERSION=1.69
+# The cargo contract version to use
+ARG CARGO_CONTRACT_VERSION=3.0.1
+# gcc package version
+ARG GCC_VERSION=4:10.2.1-1
+# wget package version
+ARG WGET_VERSION=1.21-1+deb11u1
+# g++ package version
+ARG G_VERSION=4:10.2.1-1
+# Flag to indicate whether to install cargo-contract from git
+ARG USE_GIT=false
+# Url to the cargo-contract repository to install from
+ARG CARGO_CONTRACT_GIT
```

Comment:
Can we make these just a single var here? i.e. just `CARGO_CONTRACT_GIT` and check whether it has a value or not?

Also I see they are not set anywhere yet, I guess eventually we should be able to control whether it uses a `crates.io` release or the git dependency.